TriviaServer/TriviaServer.DAL/Repositories/GameRepository.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TriviaServer.Models.Repositories;
+
+namespace TriviaServer.DAL.Repositories
+{
+    class GameRepository : IGameRepository
+    {
+        private readonly TriviaRankContext _context;
+        public GameRepository(TriviaRankContext context)
+        {
+            _context = context;
+        }
+        public async Task<Models.GameModel> CreateGame(int ownerId, string gameName, int totalQuestions, bool isPublic)
+        {
+            Game createGame = await _context.Games
+            .OrderBy(x => x.Id).LastAsync();
+
+            Game newGame = new Game
+            {
+                GameName = gameName,
+                OwnerId = ownerId,
+                StartDate = DateTime.Now,
+                TotalQuestions = totalQuestions,
+                IsPublic = isPublic
+            };
+
+            await _context.AddAsync(newGame);
+            await _context.SaveChangesAsync();
+
+            Models.GameModel appGame = new Models.GameModel
+            {
+                Id = newGame.Id,
+                GameName = newGame.GameName,
+                OwnerId = newGame.OwnerId,
+                StartDate = newGame.StartDate,
+                TotalQuestions = newGame.TotalQuestions,
+                IsPublic = newGame.IsPublic
+            };
+            return appGame;
+        }
+
+        public void EndGame(Models.GameModel appGame)
+        {
+            Game endGame = _context.Games
+                .Where(x => x.Id == appGame.Id).First();
+
+            endGame.EndDate = DateTime.Now;
+
+            _context.Add(endGame);
```

Comment:
Also, noted: this will need to be changed from Add() to Modify(), since I'm updating a record in the db.